### PR TITLE
Check and improve documentation for our tests

### PR DESCRIPTION
**Pull request by reosarevok**
The main purpose of this PR is to review and improve our test documentation, and make small fixes as relevant when something is found while reviewing the docs.

See commit messages where present for more details.

Stopping this at the end of `Controller::Artist` because it's long enough as it is.

### CODE REVIEW

**Review by yvanzo — APPROVED** [review]
Hats off to you for all of these improvements! It makes it much easier to find what is tested and what is not just by reading CI output. Plus it made you spot and fix some bugged tests! Reviewing 47 commits has been laborious, the pull request could have been split into parts, but it was certainly even more laborious to check and fix all of these tests, so thanks anyway! :ship:

**Comment by reosarevok** [comment]
Thanks! I wasn't sure when to split it, I should maybe have split Controller::Artist out of it already :) I'll try to make smaller ones next!

File: t/lib/t/MusicBrainz/Server/Controller/Artist/DeleteAlias.pm [hunk 1]
```diff
@@ -75,4 +82,37 @@ test 'Test alias deletion' => sub {
...
+        );
+    } $c;
+
+    is(@edits, 0, 'No edit was entered');
+
+    ### FIXME: This is currently broken, see MBS-12092
+    #$mech->content_contains(
+    #    'You must provide an edit note',
+    #    'Contains warning about edit note being required',
+    #);
```
[the remark this answers is not in the document]
Comment (reosarevok) [reply t/lib/t/MusicBrainz/Server/Controller/Artist/DeleteAlias.pm]: Yes :) Done now.

**Comment by reosarevok** [comment]
@brainzbot, retest this please

File: t/lib/t/MusicBrainz/Server/Controller/Artist/Aliases.pm [hunk 2]
```diff
@@ -6,57 +6,63 @@ with 't::Mechanize', 't::Context';
...
-$mech->content_unlike(qr/Test Alias/, 'other artist pages do not have the alias');
+This test checks whether artist aliases are correctly listed on the artist
+alias page, both on the site itself and on the JSON-LD data.
 
-page_test_jsonld $mech => {
-    '@context' => 'http://schema.org',
-    '@type' => 'MusicGroup',
-    '@id' => 'http://musicbrainz.org/artist/60e5d080-c964-11de-8a39-0800200c9a66',
-    'name' => 'Empty Artist'
-};
```
Comment (mwiencek) [inline t/lib/t/MusicBrainz/Server/Controller/Artist/Aliases.pm:58]: The one in `Artist::Show` is more complete, but it's useful to have an incomplete one too, to test what happens when fields are missing or empty.  So I'd prefer just moving it to `Artist::Show`.

Comment (reosarevok) [reply t/lib/t/MusicBrainz/Server/Controller/Artist/Aliases.pm:58]: Can you send a quick commit putting in there to master then? (I merged this already)

**Pull request merged**